hack/jenkins/upload-to-gcs.sh

```diff
@@ -151,15 +151,22 @@ function update_job_result_cache() {
 
   mkdir -p ${tmp_results%/*}
 
+  # Construct a valid json file
```

Comment:
Not really shell here.  Pure jq.
```
$ jq -r '.JOB[] | select(.result == "SUCCESS") | select(.version != "") | [.version,.buildnumber] | "\(.[0]) \(.[1])"' /tmp/out.json
v1.4.0-alpha.2.793+a4acf0874763b9 7478
v1.4.0-alpha.2.795+16c71aad4a2266 7479

```
Where /tmp/out.json is:
```
{"JOB" : [
{"version": "v1.4.0-alpha.2.793+a4acf0874763b9", "buildnumber": "7478", "result": "SUCCESS"},
{"version": "v1.4.0-alpha.2.795+16c71aad4a2266", "buildnumber": "7479", "result": "SUCCESS"},
{}
]}
```

And...
```
$ jq -r '.JOB[] | select(.result == "SUCCESS") | select(.version != "") | [.version,.buildnumber] | "\(.[0]) \(.[1])"' /tmp/out2.json
jq: error: Cannot iterate over null
parse error: Expected value before ',' at line 1, column 94
```

Where /tmp/out2.json is:
```
{"version": "v1.4.0-alpha.2.793+a4acf0874763b9", "buildnumber": "21513", "result": "SUCCESS"},
{"version": "v1.4.0-alpha.2.793+a4acf0874763b9", "buildnumber": "21514", "result": "SUCCESS"},
```